### ISSUE TITLE
create proper documentation

### ISSUE THREAD

**Issue by kylerbrown** [kylerbrown/bark#14]
ie [sphinx](http://www.sphinx-doc.org/en/stable/) and [readthedocs](https://readthedocs.org/).

Spyking Circus has nice docs: https://spyking-circus.readthedocs.io/en/0.4.3/

Let's copy them.

**Comment by gfetterman** [comment]
Yes!

**Comment by gfetterman** [comment]
I set up Sphinx on a [new branch](https://github.com/gfetterman/bark/tree/doc-sphinx). I also rewrote the docstrings for the `bark` module so we could see how it works.

I haven't done all of the work necessary to bring us up to spyking-circus-quality docs, but I figured it was worth pausing here to get some feedback.

Check it out, see how you like it. Let's talk a bit about whether we like this and where to go from here if we do.

ETA: if you want to clone the branch and view the html docs, you'll need to:
1. install `Sphinx` and `recommonmark`, both via pip
2. run `make html` in the `docs` directory

They'll be in `doc/_builds/html/index.html`
